src/backend/gl/src/lib.rs

```diff
@@ -70,7 +70,7 @@ impl hal::Backend for Backend {
     type ComputePipeline = native::ComputePipeline;
     type GraphicsPipeline = native::GraphicsPipeline;
     type PipelineLayout = native::PipelineLayout;
-    type DescriptorSetLayout = native::DescriptorSetLayout;
+    type DescriptorSetLayout = Vec<pso::DescriptorSetLayoutBinding>;
```

Comment:
if you do need it to just be a vec, let's do `pub type DescriptorSetLayout = Vec<pso::...>;` in `native` module and use here and elsewhere consistently